File: code/widgets/org.eclipse.scout.jswidgets.ui.html/src/main/js/jswidgets/formfield/FieldStyleLookupCall.js
/*******************************************************************************
 * Copyright (c) 2017 BSI Business Systems Integration AG.
 * All rights reserved. This program and the accompanying materials
 * are made available under the terms of the Eclipse Distribution License v1.0
 * which accompanies this distribution, and is available at
 * http://www.eclipse.org/org/documents/edl-v10.html
 *
 * Contributors:
 *     BSI Business Systems Integration AG - initial API and implementation
 ******************************************************************************/
jswidgets.FieldStyleLookupCall = function() {
  jswidgets.FieldStyleLookupCall.parent.call(this);
};
scout.inherits(jswidgets.FieldStyleLookupCall, scout.StaticLookupCall);

jswidgets.FieldStyleLookupCall.prototype._data = function() {
  return jswidgets.FieldStyleLookupCall.DATA;
};

jswidgets.FieldStyleLookupCall.DATA = [
  [scout.FormField.FieldStyle.CLASSIC, 'Classic'],
  [scout.FormField.FieldStyle.ALTERNATIVE, 'Alternative']
];

File: code/widgets/org.eclipse.scout.widgets.client/src/main/java/org/eclipse/scout/widgets/client/ui/template/formfield/FieldStyleLookupCall.java
/*
 * Copyright (c) 2010, 2023 BSI Business Systems Integration AG
 *
 * This program and the accompanying materials are made
 * available under the terms of the Eclipse Public License 2.0
 * which is available at https://www.eclipse.org/legal/epl-2.0/
 *
 * SPDX-License-Identifier: EPL-2.0
 */
package org.eclipse.scout.widgets.client.ui.template.formfield;

import java.util.ArrayList;
import java.util.List;

import org.eclipse.scout.rt.client.ui.form.fields.IFormField;
import org.eclipse.scout.rt.platform.classid.ClassId;
import org.eclipse.scout.rt.shared.services.lookup.ILookupRow;
import org.eclipse.scout.rt.shared.services.lookup.LocalLookupCall;
import org.eclipse.scout.rt.shared.services.lookup.LookupRow;

@ClassId("f3f8c108-0038-4441-b692-18c33a01d3df")
public class FieldStyleLookupCall extends LocalLookupCall<String> {

  private static final long serialVersionUID = 1L;

  @Override
  protected List<? extends ILookupRow<String>> execCreateLookupRows() {
    List<ILookupRow<String>> rows = new ArrayList<>();
    rows.add(new LookupRow<>(IFormField.FIELD_STYLE_CLASSIC, "classic"));
    rows.add(new LookupRow<>(IFormField.FIELD_STYLE_ALTERNATIVE, "alternative"));
    return rows;
  }
}
